common/persistence/telemetry/gowrap_template

```diff
@@ -0,0 +1,38 @@
+import(
+    "time"
+)
+
+{{ $decorator := (or .Vars.DecoratorName (printf "telemetry%s" .Interface.Name)) }}
+
+// {{$decorator}} implements {{.Interface.Name}} interface instrumented with retries.
+type {{$decorator}} struct {
+    {{.Interface.Type}}
+    tracer trace.Tracer
+}
+
+// new{{upFirst $decorator}} returns {{$decorator}}.
+func new{{upFirst $decorator}} (base {{.Interface.Type}}, tracer trace.Tracer) {{$decorator}} {
+    return {{$decorator}} {
+        {{.Interface.Name}}: base,
+        tracer: tracer,
+    }
+}
+
+{{range $method := .Interface.Methods}}
+    {{if $method.AcceptsContext}}
+        {{ $methodIdent := (printf "%s.%s" $.Interface.Name $method.Name) }}
+        // {{$method.Name}} wraps (printf "%s.%s" $.Interface.Name $method.Name).
+        func (_d {{$decorator}}) {{$method.Declaration}} {
+            ctx, span := _d.tracer.Start(ctx, "{{ printf "persistence.%s/%s" $.Interface.Name $method.Name }}")
+            defer span.End()
+
+            {{$method.ResultsNames}} = _d.{{$.Interface.Name}}.{{$method.Call}}
+            {{- if $method.ReturnsError}}
+            if err != nil {
+              span.RecordError(err)
+            }
+            {{end}}
+            return
```

Comment:
I'm keeping it simple for now. Just tracking the span, no extra data yet. That comes next.

This approach will work for many use cases, but not all of them. Specifically, if we got here from a RPC, this span will automatically be linked to that span. Future work will need to ensure everything has the correct Workflow ID tag, if there is no RPC.